crypto/ffc/ffc_key_generate.c

```diff
@@ -25,11 +25,11 @@ int ossl_ffc_generate_private_key(BN_CTX *ctx, const FFC_PARAMS *params,
     int ret = 0, qbits = BN_num_bits(params->q);
     BIGNUM *m, *two_powN = NULL;
 
-    /* Deal with the edge case where the value of N is not set */
-    if (N == 0)
-        N = qbits;
+    /* Deal with the edge cases where the value of N and/or s is not set */
     if (s == 0)
-        s = N / 2;
+        s = qbits / 2;
```

Comment:
s is not qbits / 2. You should probably use `s = ossl_ifc_ffc_compute_security_bits(BN_num_bits(params->p);`

Reply:
I'm not sure s should actually be a parameter to this function, I think we should always calculate it.

Reply:
I do not think so because s really depends on the parameters - whether they use safe prime or the FIPS generators. But we should perhaps just error out if `s == 0`.